app-frontend/src/app/pages/user/settings/profile/profile.module.js

```diff
@@ -72,9 +78,45 @@ class ProfileController {
 
     getCurrentUser() {
         this.authService.getCurrentUser().then(resp => {
+            this.currentUser = resp;
+            this.currentUserBuffer = _.cloneDeep(this.currentUser);
             this.provider = this.providers.find(provider => {
                 return resp.id.includes(provider.provider);
             }) || this.defaultProvider;
+            this.showProfileSettingsButton = this.isThirdParty();
+        });
+    }
+
+    isThirdParty() {
+        return this.provider.provider === 'google-oauth2' ||
+            this.provider.provider === 'twitter' ||
+            this.provider.provider === 'facebook' ||
+            this.provider.provider === 'github';
+    }
+
+    getButtonText() {
+        if (this.saved) {
+            return 'Saved';
+        }
+        if (this.error) {
+            return 'Failed';
```

Comment:
I'd make it read "Error, retry?" or something, and leave the message like that until they click it. Timed events are easy to miss.